service/model/config.go

```diff
@@ -0,0 +1,139 @@
+package model
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	errOrganizationNameRequired        = errors.New("organization name is required")
+	errDefaultReleaseMsgRequired       = errors.New("default release message is required")
+	errIntegrationMissingToken         = errors.New("token is required to enable integration")
+	errIntegrationTokenSetWhenDisabled = errors.New("token cannot be set when integration is disabled")
+)
+
+type Config struct {
+	OrganizationName      string
+	DefaultReleaseMessage string
+	Slack                 IntegrationConfig
+	Github                IntegrationConfig
+}
+
+type ConfigUpdate struct {
+	OrganizationName  *string
+	DefaultReleaseMsg *string
+	Slack             IntegrationConfigUpdate
+	Github            IntegrationConfigUpdate
+}
+
+func (c *Config) Update(u ConfigUpdate) error {
+	if u.OrganizationName != nil {
+		c.OrganizationName = *u.OrganizationName
+	}
+	if u.DefaultReleaseMsg != nil {
+		c.DefaultReleaseMessage = *u.DefaultReleaseMsg
+	}
+
+	if err := c.Slack.Update(u.Slack); err != nil {
+		return fmt.Errorf("slack: %w", err)
+	}
+	if err := c.Github.Update(u.Github); err != nil {
+		return fmt.Errorf("github: %w", err)
+	}
+
+	return c.Validate()
+}
+
+func (c *Config) Validate() error {
+	if c.OrganizationName == "" {
+		return errOrganizationNameRequired
+	}
+
+	if c.DefaultReleaseMessage == "" {
+		return errDefaultReleaseMsgRequired
+	}
+
+	return nil
+}
+
+func ToConfig(
+	orgName,
+	rlsMsg string,
+	slackEnabled bool,
+	slackToken string,
+	githubEnabled bool,
+	githubToken string,
+) (Config, error) {
+	slack, err := ToIntegrationConfig(slackEnabled, slackToken)
+	if err != nil {
+		return Config{}, fmt.Errorf("slack: %w", err)
+	}
+	github, err := ToIntegrationConfig(githubEnabled, githubToken)
+	if err != nil {
+		return Config{}, fmt.Errorf("github: %w", err)
+	}
+
+	c := Config{
+		OrganizationName:      orgName,
+		DefaultReleaseMessage: rlsMsg,
+		Slack:                 slack,
+		Github:                github,
+	}
+
+	if err := c.Validate(); err != nil {
+		return Config{}, err
+	}
+
+	return c, nil
+}
+
+type IntegrationConfig struct {
+	Enabled bool
+	Token   string
+}
+
+type IntegrationConfigUpdate struct {
+	Enabled *bool
+	Token   *string
+}
+
+func (c *IntegrationConfig) Update(u IntegrationConfigUpdate) error {
+	if u.Enabled != nil {
+		c.Enabled = *u.Enabled
+	}
+	if u.Token != nil {
+		c.Token = *u.Token
+	}
+
+	// if integration is disabled, clear the token
+	if !c.Enabled {
+		c.Token = ""
```

Comment:
Is it documented? Will it be visible and expected by user?
If not, I wouldn't complicate it here and don't do anything...